test/ecto/adapters/mysql_test.exs

```diff
@@ -166,6 +166,13 @@ defmodule Ecto.Adapters.MySQLTest do
     assert all(query) == ~s{SELECT TRUE FROM `schema` AS s0 LIMIT 3 OFFSET 5}
   end
 
+  test "aggregate filters" do
+    query = Schema |> select([r], count(r.x) |> filter(r.x > 10)) |> normalize
+    assert_raise Ecto.QueryError, fn ->
```

Comment:
Let's do a quick assertion on the message too.

Reply:
Hmm... after making the change to make it not use `varargs` anymore, the mysql adapter no longer throws any errors during planning. I assume that it would end up error-ing in the db w/ something about the filter function not existing. Should I add some manual intervention there in the mysql adapter to prevent filter from being used?

Reply:
Yes, you need to check it in the MySQL adapter and raise accordingly. You should find examples there.